include/picongpu/plugins/Emittance.hpp

```diff
@@ -799,29 +799,11 @@ namespace picongpu
                 commGather
             ));
 
-            auto y_offsets2 = y_offsets;
-            // permute ranks for ascending y offset order
-            // example:
-            //   gatherRank = {  2,  3,  4,  0,  1 }
-            //   y_off      = { 11, 15, 17, 19,  9 }
-            sort_by_key( y_offsets, gatherRanks );
-            sort_by_key( y_offsets2, y_sizes );
-
-           /* if (gatherRank ==7){
-                std::cout << "gCount_e->getHostBuffer( ).getBasePointer( ) if gatherRank = 7 ";
-                for (int i = 0; i < y_siz;i++){
-                    std::cout << gCount_e->getHostBuffer( ).getBasePointer( )[i] << " ";
-                }
-                std::cout << std::endl;
-            }
-            if (gatherRank ==10){
-                std::cout << "gCount_e->getHostBuffer( ).getBasePointer( ) if gatherRank = 10 ";
-                for (int i = 0; i < y_siz;i++){
-                    std::cout << gCount_e->getHostBuffer( ).getBasePointer( )[i] << " ";
-                }
-                std::cout << std::endl;
-            }
-            */
+            auto gatherRanks2 = gatherRanks;
+
+            sort_by_key( gatherRanks, y_offsets );
```

Comment:
I am not sure if this change is correct. Why do you sort by `rank`?

Reply:
OK it looks this is correct but by MPI definition the array is already sorted by rank therefor also `gatherRanks` is not needed. http://www.mcs.anl.gov/research/projects/mpi/mpi-standard/mpi-report-1.1/node69.htm

@ax3l could you please have a view on it because as I know you wrote it.

Reply:
mhh I am confused again: Could you please add comments above the section what you reduce in the different steps.

Reply:
@kossag14 pls add the comment and update it.
we sort the offsets by rank now, since `Gather` receives its data by rank and this data needs to be placed at a certain offset in `*dest`.

Reply:
@psychocoderHPC just annotated offline that they are now already sorted, we should therefore be able to even skip this :-)

Reply:
Is the sorting now needed or not?